src/org/benf/cfr/reader/bytecode/analysis/opgraph/op4rewriters/EnumClassRewriter.java

```diff
@@ -416,8 +434,17 @@ boolean isValid() {
         // Return of new anonymous array.
         if (!(testArr instanceof StructuredReturn)) return null;
         Expression rv = ((StructuredReturn)testArr).getValue();
-        if (!(rv instanceof NewAnonymousArray)) return null;
```

Comment:
I think you could just swap rv for a temp NewAnonymousArray if it matched the newObjectArray tests below here, and the rest of the change wouldn't be needed - need to go through your change though as I think it makes stuff clearer anyway.

Reply:
Do you mean replace `rv`'s value with a 'fake' `NewAnonymousArray` in case it matches `new EnumType[0]`, and then revert the changes this pull request makes to `CollectedEnumData`?
Do you still want this to be changed, or is it fine like this?

Reply:
Nah, I was just thinking out loud.  Merged this, thanks.